### ISSUE TITLE
Discussion

### ISSUE THREAD

**Issue by jxzhanggg** [shuuki4/Batch-Normalization#1]
Hi,
I doubt about if your code really work on dealing with training and testing differently.
Once the theano function are built, the set_runmode() method , 0 for training , 1 for testing, will not 
affect the calculating when running. The self.run_mode will always be 0.

I fix all the random seed firstly. Then, I removed the all the script to set_runmode. It get totally same result as not.

I think your code to decay learning rate and dealing with dropout suffer from the same problem as above.

**Comment by shuuki4** [comment]
@jxzhanggg 
Yes, I totally agree that your suggestion is correct. Theano builds a computation graph at first, (at the `__init__` in the BatchNormazliation.py code) so just changing the mode to 1 won't make any difference to the calculation. I wasn't completely aware about the operation method of Theano when I wrote this code. Thank you for your point.

To fix the problem at test time, code should be changed to initialize new graph, with weights restored from the trained model. Same method should be applied to dropout class, too.

I reviewed my codes after your issue, and I found SO many problems, including problems you suggested & other additional problems :( I'll take some time to fix these issues.

**Comment by jxzhanggg** [comment]
@shuuki4 
Thanks for your reply,  I'm new to theano and this problem quite confused me at first. I'm so happy that you could confirm my idea. Looking forward your new version with these bugs fixed :) and still thanks for your jobs.
